### PR TITLE
on Stage Edit, show environment for each deploy group and sort them

### PR DESCRIPTION
**Pull request by jonmoter**
/cc @zendesk/runway 

### Risks
 - None

### CODE REVIEW

File: app/views/stages/_fields.html.erb [hunk 1]
```diff
@@ -28,7 +28,7 @@
       <%= form.label :name, 'Deploy Groups', class: 'col-lg-2 control-label' %>
       <div class="col-lg-4">
         <%= form.collection_select(
```
Comment (grosser) [inline app/views/stages/_fields.html.erb]: might be simpler to move the lookup here so we don't have to repeat logic in 2 placed + break when that logic gets out of sync

Comment (jonmoter) [inline app/views/stages/_fields.html.erb]: As a rule of thumb, I dislike seeing DB queries like `DeployGroup.includes(:environment).order(:environment_id, :name).all` in the view code.

Comment (grosser) [inline app/views/stages/_fields.html.erb]: no need to add includes anymore ;)

Comment (grosser) [inline app/views/stages/_fields.html.erb]: then maybe a helper like `has_deploy_group?`

Comment (grosser) [inline app/views/stages/_fields.html.erb]: we are triggering all kinds of queries in the view layer, so a strict
separation does not exist anyway. This is just filling a select, maybe
should even be cached, so I would be easier to handle in the view.

Comment (jonmoter) [inline app/views/stages/_fields.html.erb]: Why wouldn't I want to add the `includes(:environment)`?  Is that something that's now handled by that gem you added?

Comment (grosser) [inline app/views/stages/_fields.html.erb]: exactly :)

**Comment by jonmoter** [comment]
There we go, @grosser, better?

**Comment by grosser** [comment]
:+1:

**Pull request merged**